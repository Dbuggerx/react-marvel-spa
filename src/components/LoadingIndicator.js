// @flow

import React from 'react';

const LoadingIndicator = () => (<h1>Loading...</h1>);

LoadingIndicator.displayName = 'LoadingIndicator';

export default LoadingIndicator;
