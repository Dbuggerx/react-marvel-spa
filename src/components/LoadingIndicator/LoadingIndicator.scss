@import '../scss/variables';

.loading-indicator {
  flex: 0 0 auto;
  background: map-get($colors, primary-bg);
  color: map-get($colors, primary-fg);
  padding: 15px;
  box-sizing: border-box;
  text-align: center;
}
