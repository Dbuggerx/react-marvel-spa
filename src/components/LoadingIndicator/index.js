// @flow

import React from 'react';
import type { StatelessFunctionalComponent } from 'react';
import './LoadingIndicator.scss';

/**
 * Renders a simple loading message
 */
const LoadingIndicator: StatelessFunctionalComponent<*> = () => (
    <div className="loading-indicator">
        <h1>Loading...</h1>
    </div>
);

LoadingIndicator.displayName = 'LoadingIndicator';

export default LoadingIndicator;
