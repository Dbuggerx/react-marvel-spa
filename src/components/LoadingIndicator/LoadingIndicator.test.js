import React from 'react';
import { shallow } from 'enzyme';
import LoadingIndicator from './index';

describe('LoadingIndicator', () => {
    test('renders properly', () => {
        const wrapper = shallow(<LoadingIndicator />);
        expect(wrapper).toMatchSnapshot();
    });
});
