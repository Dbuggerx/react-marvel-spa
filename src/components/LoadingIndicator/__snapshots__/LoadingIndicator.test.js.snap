// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders properly 1`] = `
<div
  className="loading-indicator"
>
  <h1>
    Loading...
  </h1>
</div>
`;

exports[`LoadingIndicator renders properly 1`] = `
<div
  className="loading-indicator"
>
  <h1>
    Loading...
  </h1>
</div>
`;
